src/sentry/api/endpoints/organization_member_index.py

```diff
@@ -40,16 +41,18 @@ class OrganizationMemberIndexEndpoint(OrganizationEndpoint):
 
     @transaction.atomic
     def save_team_assignments(self, organization_member, teams):
-        # teams may be empty
-        OrganizationMemberTeam.objects.filter(
-            organizationmember=organization_member).delete()
-        OrganizationMemberTeam.objects.bulk_create(
-            [
-                OrganizationMemberTeam(
-                    team=team, organizationmember=organization_member)
-                for team in teams
-            ]
-        )
+        lock = locks.get('org:member:{}'.format(organization_member.id), duration=5)
```

Comment:
You need this lock outside of the transaction. Otherwise it's not really doing anything.

Reply:
thanks, that makes sense